ios/FluentUI/People Picker/ContactCollectionView.swift

```diff
@@ -0,0 +1,161 @@
+    @objc public var contactList: [PersonaData] = [] {
+        didSet {
+            if (oldValue.count == 0 && contactList.count > 0) || (oldValue.count > 0 && contactList.count == 0) {
+                setupHeightConstraint()
+            }
+        }
+    }
+
+    open weak var contactCollectionViewDelegate: ContactCollectionViewDelegate?
+
+    private func configureCollectionView() {
+        translatesAutoresizingMaskIntoConstraints = false
+        showsHorizontalScrollIndicator = false
+        showsVerticalScrollIndicator = false
+        backgroundColor = Colors.surfacePrimary
+        dataSource = self
+        delegate = self
+        contentInset = UIEdgeInsets(top: 0, left: Constants.leadingInset, bottom: 0, right: 0)
+    }
+
+    @objc private func setupHeightConstraint() {
+        let height = (contactList.count > 0) ? UIApplication.shared.preferredContentSizeCategory.contactHeight : 0
+        heightConstraint.constant = height
+    }
+
+    private struct Constants {
+        static let leadingInset: CGFloat = 16.0
+    }
+
+    private let layout: ContactCollectionViewLayout
+    private var widthConstraint: NSLayoutConstraint?
+    private lazy var heightConstraint: NSLayoutConstraint = {
+        let heightConstraint = heightAnchor.constraint(equalToConstant: 0.0)
+        heightConstraint.isActive = true
+        return heightConstraint
+    }()
+    private var currentTappedIndex: Int?
+}
+
+extension ContactCollectionView: ContactViewDelegate {
+    public func didTapContactView(_ contact: ContactView) {
+        if let contactCollectionViewDelegate = contactCollectionViewDelegate, let currentTappedIndex = currentTappedIndex {
+            contactCollectionViewDelegate.didTapOnContactViewAtIndex(index: currentTappedIndex, personaData: contactList[currentTappedIndex])
+        }
+    }
+}
+
+extension ContactCollectionView: UICollectionViewDataSource {
+    @objc public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return contactList.count
+    }
+
+    @objc public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ContactCollectionViewCell.identifier, for: indexPath) as! ContactCollectionViewCell
+        cell.setup(contact: contactList[indexPath.item])
+        cell.contactView.contactViewDelegate = self
+
+        return cell
+    }
+}
+
+extension ContactCollectionView: UICollectionViewDelegate {
+    // If the Contact is not fully visible in the scroll frame, scrolls the view by an offset large enough
+    // so that the tapped Contact is fully visible.
+    @objc public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard let cell = cellForItem(at: indexPath) else {
+            return
+        }
+        currentTappedIndex = indexPath.item
```

Comment:
We should be going through ContactViewDelegate to detect tap events on the ContactViews. 
ContactCollectionView should set itself as the delegate of each of its ContactViews.